### PR TITLE
Add Ontario, Canada provider

### PR DESCRIPTION
**Pull request by chimo**

### CODE REVIEW

**Comment by chimo** [comment]
Wait... so I guess this adds coverage for the whole `ca-on` region?

If so, I should probably rename a few things.
Do you have a preference on which Provider name / NetworkId should be used?

**Comment by grote** [comment]
> Wait... so I guess this adds coverage for the whole ca-on region?

Yes it does, nice eh? :)

> If so, I should probably rename a few things.

I think you should!

> Do you have a preference on which Provider name / NetworkId should be used?

That is for @schildbach to answer. I would propose `CanadaOntarioProvider` and `CA-ON` for the Network ID.

**Comment by chimo** [comment]
Awesome, thanks for confirming :)

I'll wait for @schildbach's input and update accordingly.

**Comment by schildbach** [comment]
Or perhaps just OntarioProvider?

**Comment by chimo** [comment]
Renamed to 'OntarioProvider'.

Let me know if anything else needs to be changed!

File: enabler/src/de/schildbach/pte/NetworkId.java [hunk 1]
```diff
@@ -77,5 +77,8 @@
 	SF, SEPTA,
 
 	// Australia
-	SYDNEY, MET
+	SYDNEY, HSL, NZ, SPAIN, BR, BRFLORIPA, MET,
```
Comment (schildbach) [inline enabler/src/de/schildbach/pte/NetworkId.java]: You added a couple of extra IDs here. Can you edit your PR so that it adds only ONTARIO?

**Comment by schildbach** [comment]
Thanks for your contribution. Apart from my single comment, looks good!

File: enabler/src/de/schildbach/pte/NetworkId.java (hunk 1 above)
Comment (chimo) [inline enabler/src/de/schildbach/pte/NetworkId.java]: Oops, fixed!

**Comment by schildbach** [comment]
I just tried this provider locally. Of the 5 unit tests, 3 fail. One particularly important unit test is `nearbyStationsByCoordinate()`. It generates this query string:
```
http://api.navitia.io/v1/coverage/ca-on/coords/-75.683333;45.416667/places_nearby?type[]=stop_point&distance=50000&count=0&depth=0
```
and the response is a meager
```
{"disruptions":[],"pagination":{"start_page":0,"items_on_page":0,"items_per_page":0,"total_result":5622},"links":[{"href":"http:\/\/api.navitia.io\/v1\/coverage\/ca-on\/coords\/-75.683333;45.416667\/disruptions?start_page=1","type":"next","templated":false},{"href":"http:\/\/api.navitia.io\/v1\/coverage\/ca-on\/coords\/-75.683333;45.416667\/disruptions","type":"first","templated":false}]}
```

I checked the coordinate and [it's indeed in Ottawa, Canada](https://www.google.de/maps/place/45°25'00.0"N+75°41'00.0"W/@45.4093115,-75.7373706,11.25z/data=!4m2!3m1!1s0x0:0x0?hl=de). So it should be well covered by this provider.

Question: is there actually reason to believe this Navitia region works?

**Comment by chimo** [comment]
I'm using the Ontario provider in a custom Transportr build and it seems to be working okay so far.

Can you quickly explain how you run the live tests? I'll have a closer look at the failing tests tonight.

**Comment by schildbach** [comment]
I click on it in Eclipse and tell it to "run"…

**Comment by schildbach** [comment]
Ah, I see the problem with the nearby stations is it has count=0.

**Comment by schildbach** [comment]
I fixed the nearby stations test, will push a fix shortly.

**Comment by chimo** [comment]
Okay, thank you.

I've just been using gradle so far. I'll set up Eclipse for the future.

**Comment by schildbach** [comment]
Ah yes, the live tests don't run in the normal build. That's because they would never succeed. One of the ~100 servers is always broken…

**Comment by grote** [comment]
> Can you quickly explain how you run the live tests? I'll have a closer look at the failing tests tonight.

I explained how to run the test here: https://github.com/schildbach/public-transport-enabler/pull/64#issuecomment-181502577

We should probably add this to the `README.md`.

File: enabler/test/de/schildbach/pte/live/OntarioProviderLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	public void nearbyStationsByCoordinate() throws Exception
+	{
+		final NearbyLocationsResult result = queryNearbyStations(Location.coord(45416667, -75683333));
+		print(result);
+	}
+
+	@Test
+	public void queryDepartures() throws Exception
+	{
+		final QueryDeparturesResult result = queryDepartures("NA190", false);
```
Comment (schildbach) [inline enabler/test/de/schildbach/pte/live/OntarioProviderLiveTest.java]: Is that a valid station ID? Judging from the results of nearby stations, station IDs seem all to be prefixed with `OAW:SP:`.

**Comment by schildbach** [comment]
81ce7cdc3d8f7ebbcb5a0aa0171aaa71e7c93999 fixes the nearby stations test.

File: enabler/src/de/schildbach/pte/OntarioProvider.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.pte;
+
+import de.schildbach.pte.dto.Product;
+import de.schildbach.pte.dto.Style;
+import de.schildbach.pte.dto.Style.Shape;
+
+public class OntarioProvider extends AbstractNavitiaProvider
```
Comment (schildbach) [inline enabler/src/de/schildbach/pte/OntarioProvider.java]: You may want to put your name into a JavaDoc author tag here? See the other providers if you want that.

Comment (chimo) [inline enabler/src/de/schildbach/pte/OntarioProvider.java]: Done!

**Comment by chimo** [comment]
@grote @schildbach thanks for the help regarding the live tests. As far as I can see they're all successful on my end now.

**Comment by schildbach** [comment]
I just merged this. Stephane, thanks a lot for your work on this!

**Comment by grote** [comment]
Wohoo! \o/ Congratulations @chimo! I am looking forward to your Transportr pull request! ;)

**Comment by InfinityAX** [comment]
Hello, I'm curious about why the pull request was never merged. The website for the app states support for Ottawa and Toronto, yet I don't see it on the F-Droid release of the app. I noticed that OC Transpo has deprecated the stop-based API 2.0 in favour of GTFS-RT and RSS. Is there any way I or someone from Carleton University can add OC Transpo support to the app?

**Pull request closed without merging**